src/App.jsx

```diff
@@ -1,27 +1,108 @@
-        setTimeout(() => {
-          const savedTodoList = localStorage.getItem("savedTodoList")
-          const initialTodoList = savedTodoList ? JSON.parse(savedTodoList) : []
-          resolve({ data: { todoList: initialTodoList } })
-        }, 2000)
-      });
+  //getting my tasks from airtable
+
+  const fetchData = async () => {
+    const url = `https://api.airtable.com/v0/${
+      import.meta.env.VITE_AIRTABLE_BASE_ID
+    }/${import.meta.env.VITE_TABLE_NAME}`
+    const options = {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+      },
     };
+    try {
+      const response = await fetch(url, options);
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
 
-    fetchData().then(result => {
-      setTodoList(result.data.todoList)
+      const data = await response.json()
+
+      const todos = data.records.map((todo) => {
+        const newTodo = {
+          id: todo.id,
+          title: todo.fields.title,
+        };
+        return newTodo
+      });
+      setTodoList(todos)
       setIsLoading(false)
-    });
+    } catch (error) {
+      console.log(`Error: ${response.status}`)
+    }
+  };
+
+  //post task 
+  const postTodo = async (todo) => {
+    try {
+      const airtableData = {
+        fields: {
+          title: todo.title,
+        },
+      };
+      const response = await fetch(
+        `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${
+          import.meta.env.VITE_TABLE_NAME
+        }`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+          },
+          body: JSON.stringify(airtableData),
+        }
+      );
+      if (!response.ok) {
+        const message = `Error has occurred: ${response.status}`
+        throw new Error(`Error: ${response.status}`)
+      }
+
+      const dataResponse = await response.json()
+      console.log(dataResponse)
+      return { id: dataResponse.id, title: dataResponse.fields.title }
+    } catch (error) {
+      console.log(`Error: ${response.status}`)
+    }
+  }
+
+  //delete the task 
+
+  const postDeleteTodo = async (id) => {
+    try {
+      const response = await fetch(
+        `https://api.airtable.com/v0/${
+          import.meta.env.VITE_AIRTABLE_BASE_ID
+        }/Default${id}`,
+        {
+          method: "DELETE",
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+          },
+        }
+      );
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`)
+      }
+      const data = await response.json();
+      return data
+    } catch (error) {
+      console.log(error.message)
+    }
+  };
+
+  useEffect(() => {
+    fetchData();
   }, []);
```

Comment:
The useEffect hook here needs a dependency array to avoid side effects. Since we set up articulate (the source of data has been changed), we are no longer using the localStorage here.